### DISCUSSION TITLE
ImagePullSecrets for metamonitoring image

### DISCUSSION THREAD

**Discussion by tagathi** [grafana/mimir#7401] [Help and support] [answered]
Hello,

I need to define custom image repository links for mimit-metamonitoring pods, similar to what is described [here](https://github.com/grafana/mimir/discussions/4875)
However, our private image registry requires authentication, and it seems that the configuration [here](https://github.com/grafana/mimir/blob/2500b56c8c36976bdc57a6d725546adddf0730b9/operations/helm/charts/mimir-distributed/values.yaml#L2718) does not consider the pullSecrets that are set in toplevel 'image' section.
The statefulset is created without imagePullSecrets and the image cannot be pulled.
Any ideas on this?

Thanks,
Agathi

**Comment by dimitarvdimitrov** [comment]
thanks for raising this. I think this is something missing in the chart and can be added without introducing breaking changes

**Reply by dimitarvdimitrov** [reply, marked as answer]
i opened https://github.com/grafana/mimir/issues/7406 as a feature request and will close this discussion

**Comment by tagathi** [comment]
As a workaround I added the imagePullSecretsin the service account that is used by the statefulset:
`kubectl patch sa mimir -n mimir -p '"imagePullSecrets": [{"name":"regcred" }]'`
